util/chunk/chunk.go

```diff
@@ -0,0 +1,382 @@
+			chkCol.data = append(chkCol.data, rowCol.data[start:end]...)
+			chkCol.offsets = append(chkCol.offsets, int32(len(chkCol.data)))
+		} else {
+			chkCol.ifaces = append(chkCol.ifaces, rowCol.ifaces[row.idx])
+		}
+		chkCol.length++
+	}
+}
+
+// AppendNull appends a null value to the chunk.
+func (c *Chunk) AppendNull(colIdx int) {
+	c.columns[colIdx].appendNull()
+}
+
+// AppendInt64 appends a int64 value to the chunk.
+func (c *Chunk) AppendInt64(colIdx int, i int64) {
+	c.columns[colIdx].appendInt64(i)
+}
+
+// AppendUint64 appends a uint64 value to the chunk.
+func (c *Chunk) AppendUint64(colIdx int, u uint64) {
+	c.columns[colIdx].appendUint64(u)
+}
+
+// AppendFloat32 appends a float32 value to the chunk.
+func (c *Chunk) AppendFloat32(colIdx int, f float32) {
+	c.columns[colIdx].appendFloat32(f)
+}
+
+// AppendFloat64 appends a float64 value to the chunk.
+func (c *Chunk) AppendFloat64(colIdx int, f float64) {
+	c.columns[colIdx].appendFloat64(f)
+}
+
+// AppendString appends a string value to the chunk.
+func (c *Chunk) AppendString(colIdx int, str string) {
+	c.columns[colIdx].appendString(str)
+}
+
+// AppendBytes appends a bytes value to the chunk.
+func (c *Chunk) AppendBytes(colIdx int, b []byte) {
+	c.columns[colIdx].appendBytes(b)
+}
+
+// AppendTime appends a Time value to the chunk.
+// TODO: change the time structure so it can be directly written to memory.
+func (c *Chunk) AppendTime(colIdx int, t types.Time) {
+	c.columns[colIdx].appendInterface(t)
+}
+
+// AppendDuration appends a Duration value to the chunk.
+func (c *Chunk) AppendDuration(colIdx int, dur types.Duration) {
+	c.columns[colIdx].appendDuration(dur)
+}
+
+// AppendMyDecimal appends a MyDecimal value to the chunk.
+func (c *Chunk) AppendMyDecimal(colIdx int, dec *types.MyDecimal) {
+	c.columns[colIdx].appendMyDecimal(dec)
+}
+
+// AppendEnum appends an Enum value to the chunk.
+func (c *Chunk) AppendEnum(colIdx int, enum types.Enum) {
+	c.columns[colIdx].appendNameValue(enum.Name, enum.Value)
+}
+
+// AppendSet appends a Set value to the chunk.
+func (c *Chunk) AppendSet(colIdx int, set types.Set) {
+	c.columns[colIdx].appendNameValue(set.Name, set.Value)
+}
+
+// AppendJSON appends a JSON value to the chunk.
+func (c *Chunk) AppendJSON(colIdx int, j json.JSON) {
+	c.columns[colIdx].appendInterface(j)
+}
+
+type column struct {
+	length     int
+	nullCount  int
+	nullBitmap []byte
+	offsets    []int32
+	data       []byte
+	elemBuf    []byte
+	ifaces     []interface{}
+}
+
+func (c *column) isFixed() bool {
+	return c.elemBuf != nil
+}
+
+func (c *column) isVarlen() bool {
+	return c.offsets != nil
+}
+
+func (c *column) isInterface() bool {
+	return c.ifaces != nil
+}
+
+func (c *column) reset() {
+	c.length = 0
+	c.nullCount = 0
+	c.nullBitmap = c.nullBitmap[:0]
+	c.offsets = c.offsets[:0]
+	c.data = c.data[:0]
+	c.ifaces = c.ifaces[:0]
+}
+
+func (c *column) isNull(rowIdx int) bool {
+	nullByte := c.nullBitmap[rowIdx/8]
```

Comment:
rowIdx >> 3

Reply:
For efficiency, the compiler does it automatically.
For readability, I think '/ 8' is more readable.